src/database/firebase.tsx

```diff
@@ -18,6 +19,149 @@ const config = {
   appId: APP_ID,
 };
 
-firebase.initializeApp(config);
+!firebase.apps.length ? firebase.initializeApp(config) : firebase.app();
+
+const database = firebase.firestore();
+const treeCollection = database.collection('trees');
+const commentCollection = database.collection('comments');
+const additionalCollection = database.collection('additional');
+
+/*
+ * checkID validates that this ID exists in the `trees` table.
+ */
+export const checkID = async function (id: string): Promise<boolean> {
+  await treeCollection
+    .doc(id)
+    .get()
+    .then(doc => {
+      return doc.exists;
+    })
+    .catch(error => {
+      console.log('Error getting document:', error);
+    });
+  return false;
+};
+
+/*
+ * getTree queries the `trees` table and returns a Tree if the ID is found and an empty entry otherwise.
+ */
+export const getTree = async function (id: string): Promise<Tree> {
+  await treeCollection
+    .doc(id)
+    .get()
+    .then(doc => {
+      return doc.data() as Tree;
+    })
+    .catch(error => {
+      console.log('Error getting document:', error);
+    });
+  return { id: '' } as Tree;
+};
+
+/*
+ * getAllTrees returns a Dictionary containing all entries in the `trees` table.
+ */
+export const getAllTrees = async function (): Promise<Dictionary> {
+  const response = await treeCollection.get();
+  const trees: Dictionary = {};
+  response.docs.map(doc => {
+    const tree: Tree = doc.data() as Tree;
+    trees[tree.id] = tree;
+  });
+  return trees;
+};
+
+/*
+ * setTree creates/updates an entry in the `trees` table given a Tree.
+ */
+export const setTree = async function (
+  id: string,
+  tree: Tree,
+): Promise<boolean> {
+  await treeCollection
+    .doc(id)
+    .set({ tree })
+    .then(() => {
+      return true;
+    })
+    .catch(error => {
+      console.error('Error writing document: ', error);
+    });
+  return false;
+};
+
+/*
+ * getComment queries the `comments` table and returns a Comment if the ID is found and an empty entry otherwise.
+ */
+export const getComment = async function (id: string): Promise<Comment> {
+  await commentCollection
+    .doc(id)
+    .get()
+    .then(doc => {
+      if (doc.exists) {
+        return doc.data() as Comment;
+      }
+    })
+    .catch(error => {
+      console.log('Error getting document:', error);
+    });
+  return {} as Comment;
+};
+
+/*
+ * setComment creates/updates an entry in the `comments` table given a Comment.
+ */
+export const setComment = async function (
+  id: string,
+  comment: Comment,
+): Promise<boolean> {
+  await commentCollection
+    .doc(id)
+    .set({ comment })
+    .then(() => {
+      return true;
```

Comment:
do you need a boolean here for success? if the function doesn't throw you should consider success

Reply:
made change